Extension/Configuration/Patches/MBSaveLoadPatches.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Threading.Tasks;
+using HarmonyLib;
+using TaleWorlds.Core;
+using YAPO.Global;
+
+// ReSharper disable InconsistentNaming
+// ReSharper disable UnusedType.Global
+// ReSharper disable UnusedMember.Global
+
+namespace YAPO.Configuration.Patches
+{
+    public class MBSaveLoadPatches
+    {
+        [HarmonyPatch(typeof(MBSaveLoad), "LoadSaveGameData")]
+        public static class MBSaveLoadLoadSaveGameDataPatch
+        {
+            public static void Postfix(string saveName)
+            {
+                States.CurrentSaveId = saveName;
+                Task unused = Task.Run(LoadAsync);
+            }
+
+            private static void LoadAsync()
+            {
+                SorterConfigurationManager.Instance.LoadConfigurations();
+                (States.PartySorterConfiguration, States.OtherSorterConfiguration) =
+                    SorterConfigurationManager.Instance.GetConfiguration(States.CurrentSaveId);
+            }
+        }
+
+        [HarmonyPatch(typeof(MBSaveLoad), "SaveGame")]
+        public static class MBSaveLoadSaveGamePatch
+        {
+            public static void Postfix(string saveName)
+            {
+                States.CurrentSaveId = saveName.Replace(".tmp", "");
```

Comment:
I'm sure this needs to be done but I don't know why just looking at this part of the code. A comment may actually be beneficial

Reply:
Good idea